options/m_option.c

```diff
@@ -1456,14 +1456,20 @@ static int parse_str_list_impl(struct mp_log *log, const m_option_t *opt,
     if (op == OP_TOGGLE || op == OP_REMOVE) {
         if (dst) {
             char **list = VAL(dst);
-            int index = find_list_bstr(list, param);
-            if (index >= 0) {
-                char *old = list[index];
-                for (int n = index; list[n]; n++)
-                    list[n] = list[n + 1];
-                talloc_free(old);
-                return 1;
+            int index = 0;
+            bool found = false;
+            while (index >= 0) {
```

Comment:
could be a do-while loop to better reflect the logic

Reply:
Yeah that looks better. Changed.